openassessment/xblock/test/test_submission.py

```diff
@@ -1029,6 +1046,70 @@ def test_cancelled_submission(self, xblock):
             }
         )
 
+    @scenario('data/team_submission.xml', user_id="Red Five")
+    def test_cancelled_team_submission(self, xblock):
+        team = self.setup_mock_team(xblock)
+
+        # pylint: disable=protected-access
+        xblock.runtime._services['user'] = NullUserService()
+        xblock.runtime._services['user_state'] = UserStateService()
+        xblock.runtime._services['teams'] = MockTeamsService(True)
+
+        usage_id = xblock.scope_ids.usage_id
+        xblock.location = usage_id
+        xblock.user_state_upload_data_enabled = Mock(return_value=True)
+        xblock.teams_enabled = True
+        xblock.is_team_assignment = Mock(return_value=True)
+        anonymous_user_ids_for_team = ['Red Five', 'Bob', 'Alice', 'Chris']
+        xblock.get_anonymous_user_ids_for_team = Mock(
+            return_value=anonymous_user_ids_for_team
+        )
+        team_submission = xblock.create_team_submission(
+            ('a man must have a code', 'a man must also have a towel')
+        )
+        student_submission = dict(sub_api.get_submissions(
```

Comment:
OrderedDict, unfortunately.  I tend to prefer having single-statement assignments for this rather than extra/unused variables when possible, but I am also used to working in ES6, which is a bit more readable for it, so I can split this up, I guess.